src/components/InventoryGroups/Modals/AddSystemsToGroupModal.js

```diff
@@ -0,0 +1,208 @@
+                                    <Alert
+                                        variant="warning"
+                                        isInline
+                                        title="One or more of the selected systems
+                                    already belong to a group. Adding these systems
+                                    to a different group may impact system configuration."
+                                    />
+                                </FlexItem>
+                            )}
+                            <FlexItem>
+                                <Button
+                                    key="confirm"
+                                    variant="primary"
+                                    onClick={async () => {
+                                        if (showWarning) {
+                                            setSystemSelectModalOpen(false);
+                                            setConfirmationModalOpen(true); // switch to the confirmation modal
+                                        } else {
+                                            await handleSystemAddition([
+                                                ...selected.keys()
+                                            ]);
+                                            setTimeout(
+                                                () =>
+                                                    dispatch(
+                                                        fetchGroupDetail(groupId)
+                                                    ),
+                                                500
+                                            ); // refetch data for this group
+                                            setIsModalOpen(false);
+                                        }
+                                    }}
+                                    isDisabled={noneSelected}
+                                >
+                                    Add systems
+                                </Button>
+                                <Button
+                                    key="cancel"
+                                    variant="link"
+                                    onClick={() => setIsModalOpen(false)}
+                                >
+                                    Cancel
+                                </Button>
+                            </FlexItem>
+                        </Flex>
+                    }
+                    variant="large" // required to accomodate the systems table
+                >
+                    <InventoryTable
+                        columns={prepareColumns}
+                        variant={TableVariant.compact} // TODO: this doesn't affect the table variant
+                        tableProps={{
+                            isStickyHeader: false,
+                            canSelectAll: false
+                        }}
+                        bulkSelect={{
```

Comment:
How about creating a shared bulkSelect config builder? I have seen really similar config passed to multiple components, having a it shared would benefit a lot.

Reply:
good idea! done in the last commit